src/main/java/org/terasology/structureTemplates/internal/systems/ProtectedRegionServerSystem.java

```diff
@@ -92,22 +94,23 @@ private boolean isInProtectedRegion(Collection<Vector3i> positions) {
         return false;
     }
 
-
     @ReceiveEvent(priority = EventPriority.PRIORITY_CRITICAL)
     public void onPlaceBlocks(PlaceBlocks event, EntityRef entity) {
         EntityRef instigator = event.getInstigator();
         EntityRef player = instigator.getOwner();
         if (!player.hasComponent(ClientComponent.class)) {
             return;
         }
-        if (isInProtectedRegion(event.getBlocks().keySet())) {
+
+        if (isInProtectedRegion(event.getBlocks().keySet().stream().map(JomlUtil::from).collect(Collectors.toSet()))) {
```

Comment:
This is hard to read @skaldarnar can we break that up a bit with descriptive var names?